AWS for device

camcarlton opened the pull request.

camcarlton committed:
AWS for device

dlowrey commented on sub-projects/device/src/aws.py:
```diff
@@ -0,0 +1,42 @@
+		"""
+		i = 0
+		while i < len(self.image_list):
+		    response = self.client.compare_faces(
```
This will get assigned a new value on each iteration of the loop. Meaning that if on `i=0` it finds a match, it will get overwritten on any subsequent values of `i`. This alone is not the problem, but combined with the logic following the loop it means that the result will always be if the given face matches the last face in `self.image_list`.

camcarlton replied:
I am not sure I understand. Are you talking about the response variable?

dlowrey replied:
Yes, because it is not ever used until the while loop is finished.

dlowrey commented on sub-projects/device/src/aws.py:
```diff
@@ -0,0 +1,42 @@
+			)
+
+    	if response['FaceMatches'] != []:
+        	name = self.image_list[i]
```
Like said in a comment above, this is only ever going to tell us if the last face we compare matches or not.

camcarlton replied:
This is also fixed.

dlowrey commented on sub-projects/device/src/aws.py:
```diff
@@ -0,0 +1,42 @@
+		:param img: target image for comparing
+		"""
+		i = 0
+		while i < len(self.image_list):
```
`i` is never changed. This is an infinite while loop

dlowrey replied:
So two things here:
1. The `len()` function will get called `len(self.image_list)` amount of times. I would separate it out above the while loop
2. I think a better way to express what is trying to be accomplished here in code is using a `for` loop. Since we are just iterating over each item in the `self.image_list`, and as soon as you find a match just `break`.

camcarlton replied:
The only problem with breaking is that then we no longer have a multi-face comparison in the target image. If we don't need that functionality then I agree. 
The infinite loop issue has been fixed.

dlowrey replied:
Oh you mean if there are multiple faces in `img`? Yeah then instead of `break` just build up a list and return something like `True, ["list", "of", "ppl"] if you want

dlowrey commented on sub-projects/device/src/aws.py:
```diff
@@ -0,0 +1,42 @@
+import json
+from PIL import Image
+
+class Rekgoniton(object):
```
I think this stuff is still double tabbed

dlowrey replied:
Also `Rekgonition` -> `Rekognition`

camcarlton replied:
Yea I just rewrote the file to fix that. Class is now AWS.

dlowrey commented on sub-projects/device/src/aws.py:
```diff
@@ -0,0 +1,42 @@
+				SimilarityThreshold=80,
+			)
+
+    	if response['FaceMatches'] != []:
```
a more pythonic way `if response['FaceMatches']:`

camcarlton replied:
Fixed this. If I want to know if the list is empty I can just put a  ! in front of response['FaceMatches'], right?

dlowrey replied:
Yes!

dlowrey commented on sub-projects/device/src/aws.py:
```diff
@@ -0,0 +1,42 @@
+		else:
+			return False, ''
+
+		i += 1
```
this would need to be inside of the while loop to avoid infinite loop

camcarlton replied:
Fixed.

camcarlton committed:
S3 Intergation and Suggested FIxes

Incorporated fixes suggested by the PR review. Also intergated functionallity to grab all the files from the knownfaces bucket and store it into a list.

dlowrey reviewed (commented):
This looks good so far. I would like to integrate it with the Raspberry Pi devices tonight at the meeting!

camcarlton committed:
Python Syntax

Better syntax for if statment
